nemo_skills/inference/server/model.py

```diff
@@ -793,14 +796,61 @@ def _stream_chunks(self, response):
         """
 
         for chunk in response:
-            cur_delta = chunk.choices[0].text
+            if hasattr(chunk.choices[0], "delta"):
+                cur_delta = chunk.choices[0].delta.content
+            else:
+                cur_delta = chunk.choices[0].text
 
-            if cur_delta:
-                yield {"generation": cur_delta}
+            finish_reason = getattr(chunk.choices[0], "finish_reason", None)
+            result = {"generation": cur_delta}
+            if finish_reason:
+                result["finish_reason"] = finish_reason
+                if not cur_delta:
+                    result["generation"] = ""
+            
+            yield result
 
-            stop_reason = getattr(chunk.choices[0], "stop_reason", None)
-            if stop_reason and isinstance(stop_reason, str):
-                yield {"generation": stop_reason}
+
+class AzureOpenAIModel(OpenAIModel):
+    def __init__(
+        self,
+        host: str = '127.0.0.1',
+        port: str = '5000',
+        model=None,
+        base_url=None,
+        api_key=None,
+        api_version: str = "2024-02-15-preview",
+        max_retries: int = 3,
+        initial_retry_delay: float = 2.0,
+        **kwargs,
+    ):
+        # Call BaseModel.__init__ directly to bypass OpenAIModel.__init__
+        BaseModel.__init__(self, host=host, port=port, **kwargs)
+        from openai import AzureOpenAI
+
+        if model is None:
```

Comment:
can skip the if by doing `model = model or os.getenv("NEMO_SKILLS_OPENAI_MODEL")`